pyomo/contrib/fbbt/interval.py

```diff
@@ -39,35 +39,59 @@ def mul(xl, xu, yl, yu):
 
 
 def inv(xl, xu, feasibility_tol):
-    if xl <= feasibility_tol and xu >= -feasibility_tol:
-        # if the denominator (x) can include 0, then 1/x is unbounded.
-        lb = -inf
+    """
+    The case where xl is very slightly positive but should be very slightly negative (or xu is very slightly negative
+    but should be very slightly positive) should not be an issue. Suppose xu is 2 and xl is 1e-15 but should be -1e-15.
+    The bounds obtained from this function will be [0.5, 1e15] or [0.5, inf), depending on the value of
+    feasibility_tol. The true bounds are (-inf, -1e15] U [0.5, inf), where U is union. The exclusion of (-inf, -1e15]
+    should be acceptable. Additionally, it very important to return [0, 1/xu] when xl is 0.
+    """
+    if xu <= 0 <= xl:
+        if abs(xl - xu) <= feasibility_tol:
+            raise IntervalException(f'Division by zero in inv; xl: {xl}; xu: {xu}')
```

Comment:
Is this correct?  This logic seems a bit strange: You allow for division by zero for the intervals `[-eps, eps]`, `[eps, n]`, and `[-n, eps]`, but not `[eps, -eps]`?

Reply:
I believe this behavior is reasonable, but it might be safer to return `(-inf, inf)` for this case.

Reply:
This one is tricky. I will come back to it. We definitely need to be careful here.

Reply:
I see.  That makes sense.  Would this code then be more clear if it was written as 
```python
if xu - xl <= -feasibility_tol:
    raise InfeasibleConstraintException
elif xu - xl <= feasibility_tol and min(xl, xu) <= 0 <= max(xl, xu):
    raise IntervalException(f'Division by zero in inv')
```

Reply:
Yes. I like that. I'll make that change shortly.

Reply:
Actually, I think it has to be 

```
if xu - xl <= -feasibility_tol:
    raise InfeasibleConstraintException
elif xu <= 0 <= xl:
    raise IntervalException(f'Division by zero in inv')
```

because I want `inv(-1e-7, 1e-7, feasibility_tol=1e-6)` to return `(-inf, inf)` and not raise an exception.

Reply:
I updated this part of the code.